backend/src/middleware/auth.js

```diff
@@ -9,6 +9,7 @@ const auth = async (req, res, next) => {
   if (key != undefined) {
     // removed && key.length === 11
     // Can add this rule if wanted
+    console.log('reached here')
 
     // check to see if URL is for register, since the key would not be recorded yet
     if (req.method === 'POST' && req.url.includes('/user/')) {

```

Comment:
the last failure on circle CI is probably because this line. Since `req` only has the `query` key, `req.url === undefined` so it'll throw an error like `cannot call method includes on undefined`